### PR TITLE
VE-151 Added mutliselect option for variants

### PR DESCRIPTION
**Pull request by nikitharani**
Contributes [VE-151](https://sce.myjetbrains.com/youtrack/issue/VE-151/Add-Gateway-timing-functionality)

## Changes

- Update the vehicleSelector component logic to support multi-select for variants if multipleVariant is true.
- renamed multipleSelection to multipleGates
- Added limit tags prop
- Added story for multi-select variant
- Added tests for vehicleSelector component
- Added renderTags  with customizable label truncation and tag limits to AutoComplete Component
- removed pointerEvents it restored the default behavior similar to the original MuiAutocomplete component.

## UI/UX
@Sowbhagya-ipg  Please review the UI changes

image

Example of VehicleSelector Component integrated in VIRTO.VEHICLE remix 

vehicleSelector

## Testing notes

- User should able to select multiple Variants

## Author checklist

Before I request a review:

- [x] I have reviewed my own code-diff.
- [ ] I have tested the changes in Docker / a deploy-preview.
- [x] I have assigned the PR to myself or an appropriate delegate.
- [x] I have added the relevant labels to the PR.
- [x] I have included appropriate tests.
- [x] I have checked that the Lint and Test workflows pass on Github.
- [ ] I have populated the deploy-preview with relevant test data.
- [x] I have tagged a UI/UX designer for review if this PR includes UI/UX changes.

### CODE REVIEW

File: src/Autocomplete/Autocomplete.tsx [hunk 1]
```diff
@@ -43,6 +45,36 @@ export default function Autocomplete<
...
+    getTagProps: AutocompleteRenderGetTagProps,
+    limitTags: number
+  ) {
+    const numTags = value.length;
+    return (
+      <>
+        {value.slice(0, limitTags).map((option, index) => (
+          <Chip
+            {...getTagProps({ index })}
+            key={index}
```
Comment (mattcorner) [inline src/Autocomplete/Autocomplete.tsx:60]: Can we pick a better key than index?

File: src/VehicleSelect/VehicleSelect.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ function uniqueSortedArray(values: string[]) {
 // component to select a vehicle
 function VehicleSelect({
   flexDirection = "column",
+  limitTags = 1,
```
Comment (mattcorner) [inline src/VehicleSelect/VehicleSelect.tsx:14]: Did you mean to update `VehicleSelect`? It seems from the PR description that you were working on the `VehicleSelector` component?

File: src/VehicleSelector/VehicleSelector.types.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ export type VehicleSelectorProps = {
   /**
    * Allow multiple selections
    */
-  multipleSelection?: boolean;
```
Comment (mattcorner) [inline src/VehicleSelector/VehicleSelector.types.ts:42]: This is a breaking change. Did you find a specific case where you needed to support multipleGates = true and multipleVariants = false?

Comment (mattcorner) [reply src/VehicleSelector/VehicleSelector.types.ts:42]: Ideally we would avoid a breaking change as you'll need to go and update everywhere that this is used, and target v9 instead.

If we do need to support variable combinations of `multiple` then I'd prefer that we did it in a nested object:

```ts
type Props = {
  /**
   * Allow multiple selections
   */
  multiple?: {
     /**
      * For the variant field
      */
     variant?: boolean;
     /**
      * For the gate field (if present)
      */
     gate?: boolean;
  }
}
```

**Review by mattcorner — CHANGES_REQUESTED** [review]

File: src/VehicleSelect/VehicleSelect.tsx (hunk 2 above)
Comment (nikitharani) [reply src/VehicleSelect/VehicleSelect.tsx]: Some tests were failing  for VehicleSelect without limitTag fixed that issue and removed changes from VehicleSelect component

File: src/VehicleSelector/VehicleSelector.types.ts (hunk 3 above)
Comment (nikitharani) [reply src/VehicleSelector/VehicleSelector.types.ts]: After discussing with Dave and Sowbhagya, decided to have only one prop multipleSelection for now.

File: src/Autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -58,7 +90,12 @@ export default function Autocomplete<
       getOptionLabel={option =>
         isKeyValueOption(option) ? option.value : String(option)
       }
-      sx={{ pointerEvents: readOnly ? "none" : "auto" }}
+      renderTags={
+        limitTags
```
[the remark this answers is not in the document]
Comment (mattcorner) [reply src/Autocomplete/Autocomplete.tsx:94]: Definitely it doesn't do the right thing for -1. Here I have 3 items selected, but it says I have 6.

image

**Review by mattcorner — CHANGES_REQUESTED** [review]

**Pull request merged**